// @tensorflow-models/posenet/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var mobilenet_1 = require("./mobilenet");
exports.MobileNet = mobilenet_1.MobileNet;
var decode_multiple_poses_1 = require("./multi_pose/decode_multiple_poses");
exports.decodeMultiplePoses = decode_multiple_poses_1.decodeMultiplePoses;
var decode_single_pose_1 = require("./single_pose/decode_single_pose");
exports.decodeSinglePose = decode_single_pose_1.decodeSinglePose;
var keypoints_1 = require("./keypoints");
exports.partChannels = keypoints_1.partChannels;
exports.partIds = keypoints_1.partIds;
exports.partNames = keypoints_1.partNames;
exports.poseChain = keypoints_1.poseChain;
var posenet_model_1 = require("./posenet_model");
exports.load = posenet_model_1.load;
exports.PoseNet = posenet_model_1.PoseNet;
exports.VALID_INPUT_RESOLUTION = posenet_model_1.VALID_INPUT_RESOLUTION;
var util_1 = require("./util");
exports.getAdjacentKeyPoints = util_1.getAdjacentKeyPoints;
exports.getBoundingBox = util_1.getBoundingBox;
exports.getBoundingBoxPoints = util_1.getBoundingBoxPoints;
exports.scalePose = util_1.scalePose;
